python/ql/lib/semmle/python/frameworks/Aiohttp.qll

```diff
@@ -468,6 +468,27 @@ module AiohttpWebModel {
     override string getSourceType() { result = "aiohttp.web.Request" }
   }
 
+  /**
+   * A parameter that has a type annotation of `aiohttp.web.Request`, so with all
+   * likelihood will receive an `aiohttp.web.Request` instance at some point when a
+   * request handler is invoked.
+   */
+  class AiohttpRequestParamFromTypeAnnotation extends Request::InstanceSource,
+    DataFlow::ParameterNode, RemoteFlowSource::Range
+  {
+    AiohttpRequestParamFromTypeAnnotation() {
+      not this instanceof AiohttpRequestHandlerRequestParam and
```

Comment:
Once the API graph overhaul has been merged, we should be able to change like this:
```suggestion
      not this = any(AiohttpRequestHandlerRequestParam p).track() and
```
This should remove the problem of jumping into the middle of a path.
Until then, it would be good to have a test illustrating the problem (similar to the example given, but with a proper sanitiser so it is actually an FP); then we can see the FP disappear when we make this change.

(And with the overhaul, we should also be able to make these `InstanceSource`s a fair bit nicer, I imagine.)